### PR TITLE
[SYCL][HIP] Update XFAIL and UNSUPPORTED for some tests

### PR DESCRIPTION
**Pull request by alexbatashev**
See https://github.com/intel/llvm/issues/8812 for more info

### CODE REVIEW

File: SYCL/Basic/enqueue_barrier.cpp [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 // The test is failing sporadically on Windows OpenCL RTs
 // Disabling on windows until fixed
-// UNSUPPORTED: windows
+// UNSUPPORTED: hip_amd, windows
```
Comment (npmiller) [inline SYCL/Basic/enqueue_barrier.cpp:8]: There's a PR to fix these for HIP:

* https://github.com/intel/llvm/pull/4975

Unfortunately the PR for the tests got merged first and the actual fix stayed in review, just re-requested review for it today.

Comment (alexbatashev) [reply SYCL/Basic/enqueue_barrier.cpp:8]: I'll revert these changes if your PR gets approval

**Review by npmiller — APPROVED** [review]

**Review by bader — APPROVED** [review]

File: SYCL/Basic/barrier_order.cpp [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// UNSUPPORTED: hip
```
Comment (bader) [inline SYCL/Basic/barrier_order.cpp:1]: hip or hip_amd?

Comment (npmiller) [reply SYCL/Basic/barrier_order.cpp:1]: hip, this is the same as my other comment, it's an issue in the plugin, it will also fail on nvidia

**Pull request merged**